### ISSUE TITLE
Bundles generated offline do not start running the RN app when executed (white screen on version 0.69+)

### ISSUE THREAD

**Issue by AlenToma** [microsoft/react-native-windows#10255]
### Problem Description

I have build an app `base64string-package-maker` to be able to decrypt packages so that I could use the package on other apps.

The problem is that When I build `publish/create app Package` with `release` as a target I only get blank app.

That mean when I run `Install.ps1` the app gets installed but there is no js included what so ever.

I think that the app boundle is not included somehow or maybe something else.

I do not get Any exception or error, So I have no Idea what is happning.

### Steps To Reproduce

1- create an app and then publish it with release 
2- You will get a clean app without any js

OR

1- fork my app [base64string-package-maker](https://github.com/AlenToma/base64string-package-maker)
2- publish it and see that there is no js included

### Expected Results

The app should show the js components and run as normal.

### CLI version

8.0.3

### Environment

```markdown
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i7-3820 CPU @ 3.60GHz
    Memory: 5.70 GB / 11.92 GB
  Binaries:
    Node: 14.18.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.14.15 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.17763.0, 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: AI-212.5712.43.2112.8609683
    Visual Studio: 16.11.32602.291 (Visual Studio Enterprise 2019)
  Languages:
    Java: 18.0.1.1 - C:\Program Files\Common Files\Oracle\Java\javapath\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0
    react-native: 0.69.1 => 0.69.1
    react-native-windows: 0.69.1 => 0.69.1
  npmGlobalPackages:
    *react-native*: Not Found
```

### Target Platform Version

10.0.15063

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2019

### Build Configuration

Release

### Snack, code example, screenshot, or link to a repository

[base64string-package-maker](https://github.com/AlenToma/base64string-package-maker)

**Comment by AlenToma** [comment]
Solved it by downgrading react-native to 0.67

**Comment by chrisglein** [comment]
The link to [base64string-package-maker](https://github.com/AlenToma/base64string-package-maker) doesn't work, is it a private repo?
Is your app C# or C++?

Through validating the publish of React Native Gallery on 0.69 we know that bundling for C++ in release is good. 
The end to end test app should be validating this with tests that run release with the bundle. @jonthysell can you validate that is being covered here?

**Comment by AlenToma** [comment]
I hade downgraded to 0.6.4 and it is working now so the repo work now and I changed it private.

The repo I create is `cs` with the latest react-native and latest react-native-windows.

So as long as you create one and publish it with a release it, I am sure that you will get the same problem

**Comment by jonthysell** [comment]
I can repro this on 0.69. The bundle file **is** present in the appx package, investigating to see why it isn't loading when the app starts.

**Comment by jonthysell** [comment]
Confirmed that the bundle is seen and loaded from disk, still investigating.

**Comment by jonthysell** [comment]
Confirmed that this repros with new 69 C++ apps in release mode too.

**Comment by jonthysell** [comment]
Confirmed this issue does not repro in RNW 0.68.9.

This only repros in RNW 0.69.

**Comment by jonthysell** [comment]
Still actively investigating this, spent all day Friday trying to debug what's going on. So far:

1. The JS bundle is being created successfully at build time.
2. The bundle file is included in the appx
3. The bundle file is loaded when the native app starts
4. The bundle's contents are passed off the the JS engine and executed without any errors thrown

However, the JS app doesn't start. As far as I can see, no messages are ever sent across the bridge, view managers never create any views, etc. Without any errors, you're left with an empty ReactRootControl and the JS code is "done".

This repros in debug with a local bundle (so it's not just release). It repros with both chakra and hermes. It repros with both `dev=true` or `dev=false`. 

The only remaining lead is that it **doesn't** repro with the E2ETestApp, which loads a modified version of RNTesterApp. Which is double frustrating because the E2E tests actually interact with the UI, so if the issue **did** repro, the tests would have failed and caught this.

**Comment by jonthysell** [comment]
I have confirmed via experiment that copying and using the "new RN app template" JS within the e2e test app source repros the behavior - loading bundle via metro works, loading offline bundle gives a blank app.

**Comment by jonthysell** [comment]
The bug is that the commands to render the UI don't get processed on the native side if no native modules are ever called by the JS.

**Comment by taduyde** [comment]
any update for this?

**Comment by taduyde** [comment]
Hi @jonthysell ,
the issue also happened on version: 0.67.4

**Comment by cjhines** [comment]
I'm still encountering this on 0.69.6, [where I believe the fix was published](https://github.com/microsoft/react-native-windows/compare/react-native-windows_v0.69.6...main).

Weirdly, when I build a package in Visual Studio the bug does not occur. This is still a blocker however, as it prevent our CI deployment.

**Comment by chrisglein** [comment]
> I'm still encountering this on 0.69.6, [where I believe the fix was published](https://github.com/microsoft/react-native-windows/compare/react-native-windows_v0.69.6...main).
> 
> Weirdly, when I build a package in Visual Studio the bug does not occur. This is still a blocker however, as it prevent our CI deployment.

This was only recently backported to 0.69. You're going to need something newer. The build published next week should have it (0.69.7).

**Comment by noshaid-ali** [comment]
I'm facing this blank white window error. 
"react": "19.0.0",
"react-native": "^0.77.0",
"react-native-windows": "0.77.0"
